pymc3/plots/scatterplot.py

```diff
@@ -0,0 +1,136 @@
+        using their names.
+    divergences : Boolean
+        If True divergences will be plotted in a diferent color
+    kwargs_divergence : dicts, optional
+        Aditional keywords passed to ax.scatter for divergences
+    sub_varnames : list, optional        
+        Aditional varnames passed for plotting subsets of multidimensional variables
+    Returns
+    -------
+
+    ax : matplotlib axes
+
+    """    
+    
+    if varnames is None:
+        if plot_transformed:
+            trace_dict = get_trace_dict(trace, get_default_varnames(trace.varnames, True))
+        else:
+            trace_dict = get_trace_dict(trace, get_default_varnames(trace.varnames, False))
+        if sub_varnames is None:
+            varnames = list(trace_dict.keys())
+        else:
+            varnames = sub_varnames
+    else:
+        trace_dict = get_trace_dict(trace, varnames)
+        varnames = list(trace_dict.keys())       
+
+    if text_size is None:
+        text_size = scale_text(figsize, text_size=text_size)
+
+    if kwargs_divergence is None:
+        kwargs_divergence = {}
+    
+    numvars = len(varnames)
+
+    if figsize is None:
+        figsize = (8+numvars, 8+numvars)    
+
+    if numvars < 2:
+        raise Exception('Number of variables to be plotted must be 2 or greater.')
+        
+    if numvars == 2 and ax is not None:
+        if hexbin:
+            ax.hexbin(trace_dict[varnames[0]], trace_dict[varnames[1]], mincnt=1, **kwargs)
+        else:
+            ax.scatter(trace_dict[varnames[0]], trace_dict[varnames[1]], **kwargs)
+        
+        if divergences:
+            try:
+                divergent = trace['diverging']
+                if np.any(divergent):
+                    ax.scatter(trace_dict[varnames[0]][divergent == 1], trace_dict[varnames[1]][divergent == 1], **kwargs_divergence)
+                else:
+                    print('No divergences were found.')
+            except KeyError:
+                warnings.warn('There is no divergence information in the passed trace.')
+                return ax
+
+    if ax is None:
+        _, ax = plt.subplots(nrows=numvars, ncols=numvars, figsize=figsize)
+
+        for i in range(numvars):
+            var1 = trace_dict[varnames[i]]            
+            for j in range(i, numvars):
+                var2 = trace_dict[varnames[j]]
+
+                if i == j:
+                    ax[i, j].axes.remove()
+
+                else:
+                    if hexbin:
+                        ax[j, i].hexbin(var1, var2, mincnt=1, **kwargs)
+                    else:
+                        ax[j, i].scatter(var1, var2, **kwargs)
+                    
+                    if divergences:
+                        try:
+                            divergent = trace['diverging']
+                            if np.any(divergent):
+                                ax[j, i].scatter(var1[divergent == 1], var2[divergent == 1], **kwargs_divergence)
```

Comment:
You could use the actual position of the divergence here. They are buried in the warnings, see
https://github.com/pymc-devs/pymc3/blob/master/docs/source/notebooks/Diagnosing_biased_Inference_with_Divergences.ipynb
I don't want to expose this to users, but using it here seems fine.